### PR TITLE
Prevents NaN distances in transport operator

### PR DESCRIPTION
**Pull request by valeriaRaffuzzi**
Adresses issue #75

### CODE REVIEW

**Review by Mikolaj-A-Kowalski — APPROVED** [review]
Thanks for the fix.

File: TransportOperator/transportOperatorDT_class.f90 [hunk 1]
```diff
@@ -49,6 +49,9 @@ subroutine deltaTracking(self, p, tally, thisCycle, nextCycle)
     ! Get majornat XS inverse: 1/Sigma_majorant
     majorant_inv = ONE / self % xsData % getMajorantXS(p)
 
+    ! Should never happen! Prevents NaN distances
+    if (majorant_inv /= majorant_inv) call fatalError(Here, "Majorant is 0")
+
     DTLoop:do
       distance = -log( p% pRNG % get() ) * majorant_inv
```
Comment (valeriaRaffuzzi) [inline TransportOperator/transportOperatorDT_class.f90:56]:
```suggestion
    DTLoop:do
      distance = -log( p% pRNG % get() ) * majorant_inv

      ! Should never happen! Prevents NaN distances    
      if (distance /= distance) call fatalError(Here, "Majorant is 0")
```

Is this better @Mikolaj-A-Kowalski ?

Comment (Mikolaj-A-Kowalski) [reply TransportOperator/transportOperatorDT_class.f90:56]: Or we can check if the majorant is `0` so its inverse is `Infinity`? I don't know...  
I have a feeling that here we have a small chance to get a NaN. Really somebody would have to have `0` majorant and also sample `0` from RNG to get `0 * Inf` expression.  
```suggestion
    ! Should never happen! Prevents Inf distances
    if (abs(majorant_inv) > huge(majorant_inv)) call fatalError(Here, "Majorant is 0")

    DTLoop:do
      distance = -log( p% pRNG % get() ) * majorant_inv

```

Comment (valeriaRaffuzzi) [reply TransportOperator/transportOperatorDT_class.f90:56]: Yes sounds good!

File: TransportOperator/transportOperatorHT_class.f90 [hunk 2]
```diff
@@ -94,6 +94,8 @@ subroutine deltaTracking(self, p, tally, thisCycle, nextCycle)
     ! Get majornat XS inverse: 1/Sigma_majorant
     majorant_inv = ONE / self % xsData % getMajorantXS(p)
 
+   ! Should never happen! Prevents Inf distances
+    if (abs(majorant_inv) > huge(majorant_inv)) call fatalError(Here, "Majorant is 0")
     DTLoop:do
```
Comment (valeriaRaffuzzi) [inline TransportOperator/transportOperatorHT_class.f90:99]:
```suggestion
    if (abs(majorant_inv) > huge(majorant_inv)) call fatalError(Here, "Majorant is 0")

    DTLoop:do
```

File: TransportOperator/transportOperatorHT_class.f90 [hunk 3]
```diff
@@ -148,6 +151,10 @@ subroutine surfaceTracking(self, p, tally, thisCycle, nextCycle)
       else
         sigmaT = self % xsData % getTransMatXS(p, p % matIdx())
         dist = -log( p % pRNG % get()) / sigmaT
+
+        ! Should never happen! Catches NaN distances
+        if (dist /= dist) call fatalError(Here, "Distance is NaN")
+
```
[the remark this answers is not in the document]
Comment (valeriaRaffuzzi) [reply TransportOperator/transportOperatorHT_class.f90:157]: Yes, I can make an issue. Honestly it should be the same,  but I guess for the sake of not repeating too much code in the transport operators it makes sense.

**Pull request merged**